core/src/Kokkos_Macros.hpp

```diff
@@ -563,6 +568,29 @@ static constexpr bool kokkos_omp_on_host() { return false; }
 #endif
 #endif
 
+#ifdef KOKKOS_ENABLE_OPENACC
+#ifdef KOKKOS_COMPILER_NVHPC
+#define KOKKOS_IF_ON_DEVICE(CODE)   \
+  if (__builtin_is_device_code()) { \
+    KOKKOS_IMPL_STRIP_PARENS(CODE)  \
+  }
```

Comment:
I suspect that this means the fallback to host if no device is detected won't actually work

Reply:
There seems to be a bug in `__builtin_is_device_code`, which is not allowed to be called on host (https://forums.developer.nvidia.com/t/nvc-up-to-22-7-undefined-reference-to-builtin-is-device-code-for-empty-if-target-paths/223232). Therefore, we will have to either wait for the next release or use `acc_on_device` API.